### DISCUSSION TITLE
NTP Time Settings with DHCP

### DISCUSSION THREAD

**Discussion by Braehead** [HASwitchPlate/openHASP#480] [Documentation]
Enhancement:

I think the time settings should also support the configuration of the NTP sources from the relevant DHCP parameters.  There are [currently] 3 servers that can be manually configured for static URLs.  However, you should also be able to specify a variable, such as '%DHCP3%' to pull the third NTP server details from the DHCP parameter block.

This would allow the use of internal NTP server(s) and external(s) as fall backs.

I would like to see support for at least: DHCP - the first DHCP NTP server, DHCP1 - also the first DHCP NTP server, DHCP2 - for the second, DHCP3 for the third.  Probably no need for any further choices.

Many thanks,
Adam.

**Comment by fvanroie** [comment]
It seems that this might be possible, but only either DHCP enabled or not. I don't think you can mix and match like you are proposing... but needs to be tested. All depends if the `LWIP_DHCP_GET_NTP_SRV` is set by default in Arduino Core.

https://github.com/espressif/esp-idf/blob/56123c52aaa08f1b53350c7af30c91320b352ef4/components/lwip/include/apps/esp_sntp.h#L221-L227
